src/shared/dissect-image.c

```diff
@@ -915,6 +487,31 @@ int dissect_image(
 
                 assert((uint64_t) size < UINT64_MAX/512);
 
+                r = make_partition_devname(devname, nr, &node);
+                if (r < 0)
+                        return r;
+
+                /* So here's the thing: after the main ("whole") block device popped up it might take a while
+                 * before the kernel fully probed the partition table. Waiting for that to finish is icky in
+                 * userspace. So here's what we do instead. We issue the BLKPG_ADD_PARTITION ioctl to add the
+                 * partition ourselves, racing against the kernel. Good thing is: if this call fails with
+                 * EBUSY then the kernel was quicker than us , and that's totally OK, the outcome is good for
+                 * us: the device node will exist. If OTOH our call was successful we won the race. Which is
+                 * also good as the outcome is the same: the partition block device exists, and we can use
+                 * it.
+                 *
+                 * Kernel returns EBUSY if there's already a partition by that number or an overlapping
+                 * partition already existent. */
```

Comment:
Do we know how 'reliable' this mechanism is? Would it be worth it to keep the old code as fallback in case we get a failure that is not EBUSY, and what check can we do to ensure even if this works, that everything looks the way it's supposed to?

Reply:
well, the CI is supposed to find out how stable this is. I think this new code is sooo much simpler though, and a lot more compatible with older kernels, that I'd rather not attempt to maintain fallback code that we expressly want to replace because problematic in various environments.